floats/floats.go

```diff
@@ -490,6 +490,30 @@ func MulTo(dst, s, t []float64) []float64 {
 	return dst
 }
 
+const (
+	nanBits = 0x7ff8000000000000
+	nanMask = 0xfff8000000000000
+)
+
+// NaN returns an IEEE 754 "quiet not-a-number" value with the
+// payload specified in the low 51 bits of payload.
+// The NaN returned by math.NaN has a bit pattern equal to NaN(1).
+func NaN(payload uint64) float64 {
+	payload &^= nanMask
```

Comment:
Is there a reason to have this two lines?
`math.Float64frombits(nanBits | (payload &^ nanMask))`

Reply:
Fear of parentheses? Done.